moveit_ros/moveit_servo/include/moveit_servo/servo.hpp

```diff
@@ -125,7 +125,19 @@ class Servo
    * @param The last commanded joint states.
    * @return The next state stepping towards the required halting state.
    */
-  std::pair<bool, KinematicState> smoothHalt(const KinematicState& halt_state) const;
+  std::pair<bool, KinematicState> smoothHalt(const KinematicState& halt_state);
+
+  /**
+   * \brief Updates the smoother, if one exists, by filtering with a new state.
+   * @param state The state used to update the smoothing plugin.
+   */
+  void updateSmoother(KinematicState& state);
```

Comment:
what about `doSmoothing` as in the smoother function it calls? or `applySmoothing`?
`updateSmoother` makes me think it's changing the underlying smoothing algorithm.